referral_module/api/serializers.py

```diff
@@ -0,0 +1,85 @@
+from rest_framework import serializers
+from referral_module import models
+from django.contrib.auth.models import User
+
+
+class UserSerializer(serializers.ModelSerializer):
+
+    class Meta:
+        model = User
+        fields = ('id', 'username', )
+
+
+class UserReferrerSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = models.UserReferrer
+        fields = ('user', 'key', )
+
+
+class CampaignSerializer(serializers.ModelSerializer):
+
+    class Meta:
+        model = models.Campaign
+        fields = ('id', 'key', 'description', 'is_active', 'created',)
+
+
+class CampaignDetailsSerializer(serializers.ModelSerializer):
+
+    user_referrers = serializers.SerializerMethodField()
+
+    class Meta:
+        model = models.Campaign
+        fields = (
+            'id', 'key', 'user_referrers',
+        )
+
+    def get_user_referrers(self, obj):
+        users = User.objects.all()
```

Comment:
You don't need this queryset. `user_referrer` in your loop already has a reference to the `user` object.